test/manifold/deferred_test.clj

```diff
@@ -213,7 +213,11 @@
   ;; body produces a delayed error result
   (let [ex (Exception.)]
     (is (= ex @(capture-error
-                 (d/loop [] (future-error ex)))))))
+                 (d/loop [] (future-error ex))))))
+
+  ;; destructuring works for loop parameters
+  (is @(capture-success
+        (d/loop [[a & b] [true false]] a))))
```

Comment:
Now I have a test with map restructuring and a test with list restructuring and `recur`. I didn't want to make it too complicated, since the implementation ultimately relies on clojure.core/loop doing the right thing.